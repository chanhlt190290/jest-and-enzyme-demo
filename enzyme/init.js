import raf from './temp-polyfill';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
