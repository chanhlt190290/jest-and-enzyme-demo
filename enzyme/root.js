var  jsdom = require('jsdom');
const {JSDOM}  = jsdom;

const {document}  = (new JSDOM('<html><body></body></html>')).window; 
require('babel-register')();

var exposedProperties = ['window', 'navigator', 'document'];

global.document = document;
global.window = document.defaultView;
Object.keys(document.defaultView).forEach((property) => {
  if (typeof global[property] === 'undefined') {
    exposedProperties.push(property);
    global[property] = document.defaultView[property];
  }
});

global.navigator = {
  userAgent: 'node.js'
};

documentRef = document;

