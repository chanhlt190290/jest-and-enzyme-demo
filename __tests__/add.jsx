import React from 'react';
import renderer from 'react-test-renderer';
import Add from '../src/add.jsx';
import expect from 'expect';
import {mount, configure} from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });


describe('Add', () => {
  let add;
  let onAdd;

  beforeEach(() => {
    onAdd = jest.fn();
    add = mount(<Add onAdd={onAdd} />);
  });

  it('Add snapshot', () => {
    const component = renderer.create(<Add />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('Add requires onAdd prop', () => {
    expect(add.props().onAdd).toBeDefined();
  });

  it('Add renders button', () => {
    const button = add.find('button').first();
    expect(button).toBeDefined();
  });

  it('Button click calls onAdd', () => {
    const button = add.find('button').first();
    const input = add.find('input').first();
    input.simulate('change', { target: { value: 'Name 4' } });
    button.simulate('click');
    expect(onAdd).toBeCalledWith('Name 4');
  });
});
