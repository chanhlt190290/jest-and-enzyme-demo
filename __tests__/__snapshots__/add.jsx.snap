// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Add snapshot 1`] = `
<form>
  <input
    name="name"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Add
  </button>
</form>
`;
